be/ut/dtm0_log_ut.c

```diff
@@ -605,13 +605,86 @@ static void m0_be_ut_dtm0_log_test(void)
 	M0_LEAVE();
 }
 
+
+static void m0_be_ut_dtm0_log_init_fini(void)
+{
+	struct m0_be_dtm0_log_iter iter;
+	struct m0_dtm0_clk_src     cs;
+	struct m0_be_dtm0_log     *log;
+	int rc;
+
+	m0_dtm0_clk_src_init(&cs, M0_DTM0_CS_PHYS);
+
+	rc = m0_be_dtm0_log_alloc(&log);
+	M0_UT_ASSERT(rc == 0);
+
+	rc = m0_be_dtm0_log_init(log, NULL, &cs, false);
+	M0_UT_ASSERT(rc == 0);
+
+	m0_be_dtm0_log_iter_init(&iter, log);
+	m0_be_dtm0_log_iter_fini(&iter);
+
+	m0_be_dtm0_log_fini(log);
+	m0_be_dtm0_log_free(&log);
```

Comment:
Can we change this function name to `m0_be_dtm0_log_free0` because it zeroes the pointer.

Reply:
Good point. In reality this should be handled just like m0_free and m0_free0 - where m0_free does the freeing of memory while m0_free0 macro additionally sets the caller pointer to NULL. I will make similar changes for m0_be_dtm0_log_free/m0_be_dtm0_log_free0.

Reply:
Fixed.